src/js/utils/MesosStateUtil.js

```diff
@@ -225,26 +225,25 @@ const MesosStateUtil = {
     // Find matching executor or task to construct the task path
     [].concat(framework.executors, framework.completed_executors)
         .every(function (executor) {
-          // Find app/framework executor
-          if (executor != null &&
-              (executor.id === executorID || executor.id === taskID)) {
+          if (executor.id === executorID || executor.id === taskID) {
+
+            // Use the executor task path construct if it's a "pod" / TaskGroup
+            // executor (type: DEFAULT), otherwise fallback to the default
+            // app/framework behavior.
+            if (executor.type === 'DEFAULT') {
```

Comment:
Not a blocker for PR, but would be nice if this `DEFAULT` was extracted to a constant.